L1Trigger/L1TMuonEndCap/src/PtAssignmentEngine.cc

```diff
@@ -41,27 +42,26 @@ void PtAssignmentEngine::read(const std::string& xml_dir) {
   return;
 }
 
-void PtAssignmentEngine::load(const L1TMuonEndCapForest *payload) {
-  // unsigned pt_lut_version = payload->version_;  // Why is payload->version_ always 0? - AWB 02.06.17
-  // std::cout << "ptLUTVersion_ from configuration = " << ptLUTVersion_ << ", payload->version_ = " << payload->version_ << std::endl;
-  // if (not(pt_lut_version == unsigned(ptLUTVersion_)))
-  //   { edm::LogError("L1T") << "pt_lut_version = " << pt_lut_version << ", ptLUTVersion_ = " << ptLUTVersion_; return; }
-    
+void PtAssignmentEngine::load(int pt_lut_version, const L1TMuonEndCapForest *payload) {
+  if (ptLUTVersion_ == pt_lut_version)  return;
+  ptLUTVersion_ = pt_lut_version;
+
+  edm::LogInfo("L1T") << "EMTF using pt_lut_ver: " << pt_lut_version;
+
   for (unsigned i = 0; i < allowedModes_.size(); ++i) {
     int mode = allowedModes_.at(i);
-    
+
     L1TMuonEndCapForest::DForestMap::const_iterator index = payload->forest_map_.find(mode); // associates mode to index
     if (index == payload->forest_map_.end())  continue;
-    
+
     forests_.at(mode).loadFromCondPayload(payload->forest_coll_[index->second]);
-    
-    double boostWeight_ = payload->forest_map_.find(mode+16)->second / 1000000.;  
+
+    double boostWeight_ = payload->forest_map_.find(mode+16)->second / 1000000.;
     // std::cout << "Loaded forest for mode " << mode << " with boostWeight_ = " << boostWeight_ << std::endl;
     // std::cout << "  * ptLUTVersion_ = " << ptLUTVersion_ << std::endl;
     forests_.at(mode).getTree(0)->setBoostWeight( boostWeight_ );
 
-    if (not(boostWeight_ == 0 || ptLUTVersion_ >= 6))  // Check that XMLs and pT LUT version are consistent
-      { edm::LogError("L1T") << "boostWeight_ = " << boostWeight_ << ", ptLUTVersion_ = " << ptLUTVersion_; return; }
```

Comment:
I think this should stay - we should have a check to make sure the payload from O2O is internally consistent.